backend/services/event.py

```diff
@@ -0,0 +1,176 @@
+
+        Parameters:
+            id: a valid int representing a unique event ID
+
+        Returns:
+            Event: Object with corresponding ID
+        """
+
+        # Query the event with matching id
+        entity = self._session.query(EventEntity).get(id)
+
+        # Check if result is null
+        if entity:
+            # Convert entry to a model and return
+            return entity.to_details_model()
+        else:
+            # Raise exception
+            raise EventNotFoundException(id);
+
+    def get_events_from_organization(self, id: int) -> list[EventDetails]:
+        """
+        Get all the events hosted by an organization with id
+
+        Parameters:
+            slug: a valid str representing a unique Organization slug
+
+        Returns:
+            list[EventDetail]: a list of valid EventDetails models
+        """
+
+        # Query the event with matching organization slug
+        events = self._session.query(EventEntity).filter(EventEntity.organization_id == id).all()
+        
+        # Convert entities to models and return
+        return [event.to_details_model() for event in events]
+
+    def update(self, subject: User, event: Event) -> EventDetails:
+        """
+        Update the event
+        If none found, a debug description is displayed.
+
+        Parameters:
+            event: a valid Event model
+
+        Returns:
+            EventDetails: a valid EventDetails model representing the updated event object
+        """
+        
+        # Ensure that the user has appropriate permissions to update users
+        self._permission.enforce(subject, "organization.events.manage", f"organization/{event.organization_id}")
+
+        # Query the event with matching id
+        obj = self._session.query(EventEntity).get(event.id)
+
+        # Check if result is null
+        if obj:
+            
+            # Update event object
+            obj.name=event.name
+            obj.time=event.time
+            obj.description=event.description
+            obj.location=event.location
+            obj.public=event.public
+            
+            # Save changes
+            self._session.commit()
+            
+            # Return updated object
+            return obj.to_details_model()
+        else:
+            # Raise exception
+            raise EventNotFoundException(event.id);
+
+    
+    def delete(self, subject: User, id: int) -> None:
+        """
+        Delete the event based on the provided ID.
+        If no item exists to delete, a debug description is displayed.
+
+        Parameters:
+            id: an int representing a unique event ID
+        """
+        
+        # Find object to delete
+        event = self._session.query(EventEntity).get(id)
+
+        # Ensure that the user has appropriate permissions to delete users
+        self._permission.enforce(subject, "organization.events.manage", f"organization/{event.organization_id}")

```

Comment:
Move this check above the query for a slight efficiency gain in the scenario where the user does not have permission.

Reply:
Order is actually needed here in order since we need the organization ID to run the permission check!